src/super_gradients/common/environment/environment_config.py

```diff
@@ -1,15 +1,38 @@
+import dataclasses
 import os
-
+import argparse
 import pkg_resources
 
+__all__ = ["PKG_CHECKPOINTS_DIR", "EXTRA_ARGS", "device_config"]
+
+
 try:
     PKG_CHECKPOINTS_DIR = pkg_resources.resource_filename("checkpoints", "")
 except Exception:
     os.makedirs(os.path.join(os.getcwd(), "checkpoints"), exist_ok=True)
     PKG_CHECKPOINTS_DIR = os.path.join(os.getcwd(), "checkpoints")
 
 
-DDP_LOCAL_RANK = int(os.getenv("LOCAL_RANK", default=-1))
+def _get_requested_rank():
+    if os.getenv("LOCAL_RANK") is not None:
+        return int(os.getenv("LOCAL_RANK"))
+    else:
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--local_rank", default=-1)
+        args, _ = parser.parse_known_args()
+        return args.local_rank
+
+
+_DDP_REQUESTED_RANK = _get_requested_rank()
```

Comment:
Maybe it make sense to call pop_arg("local_rank") in _get_requested_rank() instead of init_trainer?
It seems we call _get_requested_rank() only once and set _DDP_REQUESTED_RANK. We do this at import level, which would guarantee pop_arg("local_rank") to be called before hydra is about to start complaining on presence of `--local_rank`

Reply:
Mmhh the thing I'm afraid with poping on import is that a user might expect to find `--local_rank` when using `torch.distributed.launch` and if we pop it behind his back it might be very confusing for him. The advantage of using init_trainer is that we can document the function so that the user knows what happens when calling it, it wouldnt be hidden.

Maybe what we can do is as you say pop on import, and if we pop it add a log.info to let the user know.
In most cases we would'nt even log anything, because when launching with python, torchrun or torch.distributed.run the local rank is passed as env variable (LOCAL_RANK) and not as --local_rank arg